src/lib.rs

```diff
@@ -26,80 +26,98 @@
 )]
 
 use async_session::{async_trait, serde_json, Result, Session, SessionStore};
-use redis::{aio::Connection, AsyncCommands, Client, IntoConnectionInfo, RedisResult};
+use redis::{aio::ConnectionManager, Client, Cmd, FromRedisValue, RedisResult};
+use std::fmt::{Debug, Formatter};
 
 /// # RedisSessionStore
-#[derive(Clone, Debug)]
+/// This redis session store uses a multiplexed connection to redis with an auto-reconnect feature.
+#[derive(Clone)]
 pub struct RedisSessionStore {
-    client: Client,
+    /// A `ConnectionManager` that wraps a multiplexed connection and automatically reconnects to the server when necessary.
+    connection: ConnectionManager,
+    /// The prefix to be used for all session keys in Redis.
     prefix: Option<String>,
 }
 
 impl RedisSessionStore {
+    /// creates a redis store from a redis URI
+    /// ```rust
+    /// # use async_redis_session::RedisSessionStore;
+    /// let store = RedisSessionStore::from_uri("redis://127.0.0.1", None);
+    /// ```
+    pub async fn from_uri(uri: &str, prefix: Option<String>) -> RedisResult<Self> {
+        let connection = ConnectionManager::new(Client::open(uri).unwrap()).await?;
+        Ok(Self { connection, prefix })
+    }
+
     /// creates a redis store from an existing [`redis::Client`]
     /// ```rust
     /// # use async_redis_session::RedisSessionStore;
     /// let client = redis::Client::open("redis://127.0.0.1").unwrap();
-    /// let store = RedisSessionStore::from_client(client);
+    /// let store = RedisSessionStore::from_client(client, None);
     /// ```
-    pub fn from_client(client: Client) -> Self {
-        Self {
-            client,
-            prefix: None,
-        }
+    pub async fn from_client(client: Client, prefix: Option<String>) -> RedisResult<Self> {
+        Ok(Self {
+            connection: client.get_tokio_connection_manager().await?,
```

Comment:
I'm not familiar with this interface, but am surprised to see tokio in the method. This is runtime agnostic even though it says tokio?

Reply:
Oddly it compiled like that.

There is also this way: https://docs.rs/redis/latest/redis/aio/struct.ConnectionManager.html#method.new